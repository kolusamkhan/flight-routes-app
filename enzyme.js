const Adapter = require('enzyme-adapter-react-17-updated')
/**
 * Setup enzyme adapter class for testing env setup.
 */
require('enzyme').configure({adapter: new Adapter()})