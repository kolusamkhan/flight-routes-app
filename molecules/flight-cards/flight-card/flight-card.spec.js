import 'jsdom-global/register';
import * as React from 'react';
import expect from 'expect';
import { shallow, mount, render, ReactWrapper } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import FlightCard from './index';
import { NextIntlProvider, useTranslations } from 'next-intl';

global.__DEV__ = true;
jest.mock('next/router', () => ({
  useRouter: () => ({ locale: 'en' })
}));
const message = require("../../../locales/en.json");


const jsonData = require("../../../pages/api/routes/flight_schedule.json");
const flightCards = jsonData?.data?.flightRouteResponse?.routes;
const flightCard = flightCards && flightCards.length > 0 ? flightCards[0] : {};

describe('FlightCard', () => {
  describe('FlightCard component', () => {
    it('should render without throwing an error', function () {
      const wrapper = shallow(
        <NextIntlProvider messages={message}>
          <FlightCard flightCard={flightCard} index={1} />
        </NextIntlProvider>);
      expect(shallowToJson(wrapper)).toMatchSnapshot();
    })

    it('should render FlightCard content', function () {
      const wrapper = mount(<div>
        <NextIntlProvider messages={message}>
          <FlightCard flightCard={flightCard} index={1} />
        </NextIntlProvider>);
      </div>);
      expect(wrapper.find('.flight_card')).toBeTruthy();
    })
  })
})