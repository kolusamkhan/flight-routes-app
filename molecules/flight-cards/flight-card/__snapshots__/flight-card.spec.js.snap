// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlightCard FlightCard component should render without throwing an error 1`] = `
<IntlProvider
  locale="en"
  messages={
    Object {
      "content": Object {
        "common": Object {
          "nav": Object {
            "home": "Home",
            "manageBooking": "Manage Booking",
            "myAccount": "My Account",
          },
          "navFooter": Object {},
          "title": "Fly Better",
        },
        "pageFlightRoutes": Object {
          "card": Object {
            "arrivalTitle": "Arrival",
            "ctaTitle": "Select",
            "departureTitle": "Departure",
          },
          "resultsTitle": "Search Results",
          "title": "Flight Routes",
        },
        "seo": Object {},
      },
    }
  }
>
  <FlightCard
    flightCard={
      Object {
        "arrDate": "2021-09-23T00:00:00Z",
        "arrTime": "0705",
        "deptDate": "2021-09-23T00:00:00Z",
        "deptTime": "0230",
        "destination": "LHR",
        "display": Object {
          "arrDate": "2021-09-23",
          "arrTime": "07:05",
          "arrivalTerminal": "Terminal 4",
          "departureTerminal": "Terminal 3",
          "deptDate": "2021-09-23",
          "deptTime": "02:30",
          "destination": "London Heathrow (LHR)",
          "origin": "Dubai (DXB)",
        },
        "legs": Object {
          "arrSTN": "LHR",
          "arrTime": "0705",
          "blockTime": "07:35",
          "codeshare": "",
          "confg": "FJY",
          "connseq": "1888482",
          "deptSTN": "DXB",
          "deptTime": "0230",
          "flightAircraftType": "388",
          "flightDate": "2021-09-23T00:00:00Z",
          "flightNo": "7",
          "flightSEQ": "1",
          "flightType": "EK",
          "legseqno": "1",
          "localACFTType": "388Y",
          "originDate": "2021-09-23T00:00:00Z",
          "totalHours": "7.35",
        },
        "operator": "EK",
        "origin": "DXB",
        "stopCounts": 0,
        "totalBlockTime": "7.35",
      }
    }
    index={1}
  />
</IntlProvider>
`;
